### ISSUE TITLE
simplewallet loses balance..

### ISSUE THREAD

**Issue by spotme-main** [forknote/forknote#86]
Running latest binaries on Windows and Linux

I am seeing a simplewallet that does not keep its balance.  I deleted the blockchain and resynced.   upon resync balance is now 0.  Transactions were valid.

2018-Feb-04 12:34:49.870021 INFO    [simplewallet] Reset completed successfully.
2018-Feb-04 12:34:52.295688 INFO    [simplewallet] Height 4796, transaction 9155b47aea9b0bc164cc5895b264eb545e6a1508265acfc117ea8b0e609e2758, received 1000.00000000000
2018-Feb-04 12:34:52.299689 INFO    [simplewallet] Height 4824, transaction dd5d60eca0993696953b38ad94b2d5212c4fd4a0e77ce883e3f8af85256027b5, received 100.00000000000
2018-Feb-04 12:34:52.303690 INFO    [simplewallet] Height 4825, transaction f6b15a6afc5d70c32a79f1295f07d27de9dc48e7ea1ef6c922c18c425653edae, received 30.00000000000
2018-Feb-04 12:35:12.700165 INFO    [simplewallet] available balance: 1130.00000000000, locked amount: 0.00000000000

Performed procedure of deleting blockchain, resynced and then resetting again

018-Feb-04 12:47:09.406797 INFO    [simplewallet] Reset completed successfully.
2018-Feb-04 12:47:20.039246 INFO    [simplewallet] available balance: 0.00000000000, locked amount: 0.00000000000
2018-Feb-04 12:47:40.685025 INFO    [simplewallet] Wallet closed

Blockchain is relatively new.  https://www.spotmecoin.com

**Comment by pmitchev** [comment]
What version are you using?

**Comment by pmitchev** [comment]
Can you try if adding SYNC_FROM_ZERO=1 to your config file resolves the problem?

**Comment by spotme-main** [comment]
Will do

**Comment by spotme-main** [comment]
Its already there.   

`seed-node=45.32.215.48:19497
seed-node=207.148.25.69:19497
seed-node=207.148.27.162:19497
seed-node=45.76.151.90:19497
seed-node=45.32.109.221:19497
seed-node=198.13.43.137:19497
seed-node=108.61.201.86:19497
seed-node=199.247.2.56:19497
seed-node=185.92.223.60:19497
seed-node=45.77.191.179:19497


EMISSION_SPEED_FACTOR=18
DIFFICULTY_TARGET=120
CRYPTONOTE_DISPLAY_DECIMAL_POINT=11
MONEY_SUPPLY=18446744073709551615
GENESIS_BLOCK_REWARD=6087425544324152000
SYNC_FROM_ZERO=1
DEFAULT_DUST_THRESHOLD=100000
MINIMUM_FEE=100000
CRYPTONOTE_MINED_MONEY_UNLOCK_WINDOW=10
CRYPTONOTE_BLOCK_GRANTED_FULL_REWARD_ZONE=100000
MAX_TRANSACTION_SIZE_LIMIT=100000
CRYPTONOTE_PUBLIC_ADDRESS_BASE58_PREFIX=57
DIFFICULTY_CUT_V1=60
DIFFICULTY_CUT_V2=60
DIFFICULTY_CUT=0
DIFFICULTY_LAG_V1=15
DIFFICULTY_LAG_V2=15
DIFFICULTY_LAG=0
DIFFICULTY_WINDOW_V1=720
DIFFICULTY_WINDOW_V2=720
DIFFICULTY_WINDOW=17
p2p-bind-port=19497
rpc-bind-port=19498
BYTECOIN_NETWORK=20026ce4-5b11-3d8e-78de-d00070a50b04
CRYPTONOTE_NAME=SPOT
GENESIS_COINBASE_TX_HEX=010a01ff0001c0f5d1f0faa8b8bd5402346d1a3b284587e258c5451eff42b2a439c579c9cc09920897a724381cf2013f2101580d020343890753091d1146a2afaf6b412c1151ad7bcb2db23e0e3a156119bf
MAX_BLOCK_SIZE_INITIAL=100000
UPGRADE_HEIGHT_V2=1
UPGRADE_HEIGHT_V3=30
daemon-port = 19498
bind-port = 9090`

**Comment by pmitchev** [comment]
Is your wallet created with the latest version or some of the previews versions?

**Comment by pmitchev** [comment]
If the wallet is created with v2.1.1, your must export your keys and reimport them into a new wallet. V2.1.1 has a bug causing wallets created with simplewallet not to sync correctly

**Comment by spotme-main** [comment]
I was bolding  the sync_from_0

I will double check the executable.  Any possibility it could be coins were mined on an alternate chain and then lost?

**Comment by spotme-main** [comment]
running latest binary downloaded directly from forknote

**Comment by pmitchev** [comment]
That might be the case. If you export the keys and import them in a new wallet you will have a 100% answer is it a bug in the software, or an alternative chain

**Comment by spotme-main** [comment]
export and import brought the coins back the transfers made to the wallet listed above.

**Comment by fuegohd** [comment]
This is happening to me as well. I've tried on Mac, Windows and Linux same issue. One minute the balance is there next minute it's gone. Reset doesn't bring it back. Only way I've been able to recover the balance is with the import option using private keys. This is certainly a major issue but I don't know whether it's with simplewallet or the blockchain itself. 

I saw a few comments from bytecoin users complaining about balances disappearing from their wallets and today noticed they launched a new beta wallet. Maybe they solved the issue with this one but I don't see the repo yet.

**Comment by pmitchev** [comment]
@fuegohd are you using simplewallet also? Are you using the latest version of forknote? Is reset helping?

The new bytecoin repo is here:
https://github.com/bcndev/bytecoin
Of course do not use if for real money until it's vetted at least in some degree by independent entity.

**Comment by fuegohd** [comment]
Yes I just downloaded from forknote.net/download. Reset is actually what makes the balance disappear. I reset to see the new balance which should have increased and instead it empties or lowers the balance. See these screenshots: https://imgur.com/Rq79XFV https://imgur.com/5jxqdoX

This screenshot https://imgur.com/nmlA4qS shows the balance when I open the wallet at 20M, then after one reset it goes to 19M and after the another reset it reduces even further to 17k. I don't get where the coins go, so imagine this happens to a user who purchased the coins.

**Comment by pmitchev** [comment]
Was the wallet created with v2.1.2?  The way to restore your wallet now is to export the keys and reimport them into new wallet

**Comment by fuegohd** [comment]
Yes my project is fairly new so I believe 2.1.2 had been out for a while when this network launched. Yes the only way to get the balance back is to import. What I'd like to figure out it how to stop this from happening. Not sure building a new QT or Python GUI will help...

**Comment by pmitchev** [comment]
Are you using forknote or the generator?

**Comment by fuegohd** [comment]
I went into the forknote directory and removed the daemon log file. Launched daemon, opened the wallet and reset. Balance reappeared.

**Comment by spotme-main** [comment]
Why would removing a log file restore the balance?

**Comment by pmitchev** [comment]
Restarting the daemon was the thing that fixed it. I got the same issue with walletd

**Comment by fuegohd** [comment]
I had restarted the daemon several times even restarted the machine and nothing. I know the daemon wasn't syncing but reset doesn't work in the daemon and reset the wallet did nothing. The only other thing I did was start a miner on another machine maybe that did it?

**Comment by mtl1979** [comment]
Reset in wallet doesn't work for two reasons.

1. Daemon doesn't keep whole blockchain in memory... It does only keep last N blocks, this is obvious when the wallet is reset as it will rescan first few thousand blocks very fast and then slows down
2. Any changes in parameters inside config file will break wallet reset, this includes enabling of for example Zawy diff
